hazelcast/include/hazelcast/client/IAtomicLong.h

```diff
@@ -190,7 +190,7 @@ namespace hazelcast {
              * @return an {@link ICompletableFuture} with the current value
              * @since cluster version 3.7
              */
-            std::shared_ptr<ICompletableFuture<int64_t> > getAsync();
+            future<std::shared_ptr<int64_t>> getAsync();
```

Comment:
Should not these be future<int64_t> ? Do we need shared ptr's for primitives ?

Reply:
No, we don't but I wıll change it when I change SerializationService to not use shared_ptr, this pointer is due to current serializationservice `toObject` method.